loader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/ModValidator.java

```diff
@@ -72,7 +72,22 @@ public ITransformationService.Resource getPluginResources() {
     }
 
     public ITransformationService.Resource getModResources() {
-        return new ITransformationService.Resource(IModuleLayerManager.Layer.GAME, this.candidateMods.stream().map(IModFile::getSecureJar).toList());
+        Predicate<IModFile> shouldLoadResource;
```

Comment:
This is a cursed way of implementing this, I think it can be made simpler by building a `Stream<IModFile>` in each branch.